server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -200,8 +200,17 @@ public class SnapshotsService extends AbstractLifecycleComponent implements Clus
         Setting.Property.Dynamic
     );
 
+    public static final Setting<Boolean> SHALLOW_SNAPSHOT_V2 = Setting.boolSetting(
+        "snapshot.shallow_snapshot_v2",
+        false,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
```

Comment:
Is this going to be per snapshot configurable?

Reply:
will be cluster level setting. We have added an attribute in SnapshotInfo to identify snapshots v2. What do you mean when we say per snapshot configurable?

Reply:
That makes sense, clarifies